### PR TITLE
Don't show registration and add post links for private deployments

### PR DESCRIPTION
**Pull request by jasonmule**
#859

[Review on Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/60)

### CODE REVIEW

File: server/www/index.html [hunk 1]
```diff
@@ -69,7 +69,8 @@ <h1 class="beta" ng-bind="site.name"></h1>
                             <p class="header-description" ng-bind="site.description"></p>
 
                             <div class="create-admin">
-                                <a class="button button-primary" ng-href="/posts/create">
+                              <a class="button button-primary" ng-href="/posts/create"
+                                 ng-show="!site.private || site.private && loggedin">
```
Comment (rjmackay) [inline server/www/index.html]: Couldn't this just be if `loggedIn || !site.private` ? .. I'd be tempted to move this logic into a function in the controller. So you can just do `ngShow="canCreatePost()"`

File: server/www/templates/partials/user-menu.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <div class="user-admin logged-out" ng-hide="currentUser">
     <a ng-href="/login" class="user-login" translate>nav.login</a>
-    <a ng-href="/register" class="user-register" translate>nav.register</a>
+    <a ng-href="/register" class="user-register" translate ng-show="!site.private">nav.register</a>
```
Comment (rjmackay) [inline server/www/templates/partials/user-menu.html]: Similar - maybe put this in a function so its `canRegister` .. more understandable then.

**Comment by rjmackay** [comment]
Minor feedback in comments - but looks good otherwise :boom:

**Comment by rjmackay** [comment]
Looks good so far.
I think you need to redirect any non-logged in users to /login though? (on private deployments)
We could do with some frontend tests too

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/ushahidi/platform-client/60#-:-K7xAQJNwXtLBAFIpiiH:867836270)*

**Comment by jasonmule** [comment]
I think redirects to /login already happen?

**Comment by rjmackay** [comment]
Looks good.
You're right redirect to login is probably triggered when we attempt to load any content.

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/ushahidi/platform-client/60#-:-KA7TwTtMabt2hlhJtpz:-738962332)*

**Comment by willdoran** [comment]
This looks good to me, it even has a groovy spec test, which is going to conflict with the test I added yesterday :D 

My only question is whether it should merge into develop or master?

**Comment by rjmackay** [comment]
Please merge to develop

**Pull request closed without merging**